content/blog/2019/egress-traffic-control-in-istio-part-2/index.md

```diff
@@ -0,0 +1,116 @@
+firewall that will enforce that traffic from the cluster to the outside is allowed for the egress gateway only. See
+an example of the [Kubernetes network policies configuration](/docs/tasks/traffic-management/egress/egress-gateway/#apply-kubernetes-network-policies).
+
+You must also increase the security measures applied to the Istio control plane pods and the egress gateway, for example:
+
+* run them on nodes separate from the application nodes
+* run them in a separate namespace
+* apply Kubernetes RBAC and Network policies to protect them
+* monitor them more thoroughly and more frequently than the application pods
+
+After all, if the attackers are able to attack Istio Mixer or the egress gateway, they could
+violate any policy.
+
+Once you direct egress traffic through an egress gateway and apply the additional security mechanisms,
+you can securely monitor and enforce security policies for the traffic.
+
+The following diagram shows Istio's security architecture, augmented with an L3 firewall which is part of the
+[additional security mechanisms](/docs/tasks/traffic-management/egress/egress-gateway/#additional-security-considerations)
+that should be provided outside of Istio.
+
+{{< image width="80%" link="./SecurityArchitectureWithL3Firewalls.svg" caption="Istio Security Architecture with Egress Gateway and L3 Firewall" >}}
+
+You can configure the L3 firewall trivially to only allow incoming traffic through the Istio ingress gateway and
+only allow outgoing traffic through the Istio egress gateway. The Istio proxies of the gateways enforce policies and report telemetry as all other proxies in the mesh.
+
+Now let's examine possible attacks and let me show you how the secure control of egress traffic in Istio prevents them.
+
+## Preventing possible attacks
+
+Consider the following security policies for egress traffic:
+
+* Application **A** is allowed to access `*.ibm.com`, which includes all the external services with URLs matching
+`*.ibm.com`.
+* Application **B** is allowed to access `mongo1.composedb.com`.
+* All egress traffic is monitored.
+
+Suppose the attackers have the following goals:
+
+1. To access `*.ibm.com` from your cluster.
+1. To access `*.ibm.com` from your cluster, unmonitored. The attackers want their traffic to be unmonitored to prevent a
+possibility that you will detect the forbidden access.
+1. To access `mongo1.composedb.com` from your cluster.
+
+Now suppose that the attackers manage to break into one of the pods of application **A**. The attackers try to use the compromised pod to perform the forbidden access.
+
+First, let's see which attacker's goals it is possible for you to thwart:
+
+1. Since application **A** is allowed to access `*.ibm.com`, the compromised pod is able to access it. There is no way to prevent such access since there is no way to distinguish, at
+least initially, between the original and the compromised versions of the pod.
+1. Fortunately, if you can monitor all
```

Comment:
Don't force readers to go back to the previous list to understand the current one. Instead, just briefly mention the attacker's goal.
```suggestion
- Fortunately, you can monitor all access to external services, detect suspicious traffic, and thwart attackers from 
  gaining unmonitored access to `*.ibm.com`. For example, you could apply anomaly detection tools on the 
  egress traffic logs.
```